test/integration-tests/tests/jd_integration.rs

```diff
@@ -221,3 +224,98 @@ async fn jds_receive_solution_while_processing_declared_job_test() {
 
     assert!(tokio::net::TcpListener::bind(jds_addr).await.is_err());
 }
+
+// This test ensures that JDS does not exit upon receiving a `ProvideMissingTransactionsSuccess`
+// message containing a transaction set that differs from the `tx_short_hash_list`
+// in the Declare Mining Job.
+//
+// It is performing the verification by connecting to JDS after the message exchange
+// to check whether it remains alive
+#[tokio::test]
+async fn jds_handles_provide_missing_transaction_success_with_unexpected_tx_list() {
+    start_tracing();
+    let (tp_1, tp_addr_1) = start_template_provider(None);
+
+    let (tp_2, tp_addr_2) = start_template_provider(None);
+
+    assert!(tp_2.fund_wallet().is_ok());
+    assert!(tp_2.create_mempool_transaction().is_ok());
+
+    let (_pool, pool_addr) = start_pool(Some(tp_addr_1)).await;
+    let (_jds, jds_addr) = start_jds(tp_1.rpc_info()).await;
+
+    let provide_missing_transaction_success_replace = ReplaceMessage::new(
+        MessageDirection::ToUpstream,
+        MESSAGE_TYPE_PROVIDE_MISSING_TRANSACTIONS_SUCCESS,
+        AnyMessage::JobDeclaration(
+            roles_logic_sv2::parsers::JobDeclaration::ProvideMissingTransactionsSuccess(
+                ProvideMissingTransactionsSuccess {
+                    request_id: 1,
+                    transaction_list: Seq064K::new(Vec::new()).unwrap(),
+                },
+            ),
+        ),
+    );
+
+    // This sniffer sits between `jds` and `jdc`, replacing `ProvideMissingTransactionSuccess`
+    // with `ProvideMissingTransactionSuccess`.
+    let (sniffer_a, sniffer_a_addr) = start_sniffer(
```

Comment:
There is only a single sniffer in the test, so this can be just (sniffer, sniffer_addr)